src/base/settingsstorage.h

```diff
@@ -62,4 +62,27 @@ public slots:
     mutable QReadWriteLock m_lock;
 };
 
+template<typename T>
+class StorableValue
+{
+public:
```

Comment:
>This class has an inconsistency

Agree. How about like this:
```c++
void set(const T &value) const
T operator() (const QVariant &defaultValue = QVariant()) const  // get
```

>But, IMO, the default value is life-time constant for object of this class.

This will depend on its scope, for example, see `storeDialogSize`.
It's possible to expand the template like this:
```c++
template<typename T>
class StorableValue
{
public:
    explicit StorableValue(const QString &keyName, const QVariant &defaultValue = QVariant())
        : m_keyName(keyName)
        , m_defaultValue(defaultValue)
    {
    }

    void set(const T &value) const
    {
        // set
        SettingsStorage::instance()->storeValue(m_keyName, value);
    }

    T operator() (const QVariant &defaultValue = QVariant()) const
    {
        // get
        QVariant defValue = defaultValue.isValid() ? defaultValue : m_defaultValue;
        return SettingsStorage::instance()->loadValue(m_keyName, defValue).template value<T>();
    }

private:
    QString m_keyName;
    QVariant m_defaultValue;
};
```